etrago/tools/utilities.py

```diff
@@ -702,6 +707,31 @@ def load_shedding(self, temporal_disaggregation=False, **kwargs):
             "Generator",
         )
 
+        if negative_load_shedding:
+            # Add negative load shedding generators for AC buses
+            ac_buses = network.buses[network.buses.carrier == "AC"].index
```

Comment:
This way the negative load shedding can not solve the problems for the EV buses. Could you add the EV buses here?

Reply:
Right. I made some changes in 97642abbea11b3e99a3cc425e81099730dd61e8e to make the allocation of the negative load shedding carrier flexible.